src/main/java/de/bonndan/nivio/output/map/hex/GroupAreaFactory.java

```diff
@@ -26,95 +28,112 @@
      * <p>
      * There is clearly much room for improvement here. It's only that I haven't found a better approach so far.
      *
-     * @param occupied       tiles occupied by items
-     * @param group          the group
-     * @param allVertexHexes a mapping from item to its hex (all, unfiltered)
+     * @param hexesToItems tiles occupied by items
+     * @param group        the group
      * @return all hexes the group consists of (an area)
      */
-    public static Set<Hex> getGroup(Set<Hex> occupied, Group group, Map<Item, Hex> allVertexHexes) {
+    public static Set<Hex> getGroup(BidiMap<Hex, Object> hexesToItems, Group group) {
```

Comment:
Which special functionality provided by the BidiMap data structure are we using? I have only seen `getKey()` and `containsKey()`. If that is so, why isn't that just the value in a normal Map structure?

Reply:
Excellent observation. It has been refactored.